src/components/Popups/ReplyField/ReplyField.tsx

```diff
@@ -77,37 +112,64 @@ export default class ReplyField extends React.Component<Props, State> {
 
         const range = selection.getRangeAt(0);
         const textNode = range.endContainer;
+        const currentCursor = range.endOffset;
+
+        const { mentionString, mentionTrigger } = activeMention;
+        const offset = mentionString.length + mentionTrigger.length;
+        const mentionStart = currentCursor - offset;
 
-        range.setStart(textNode, activeMention.start);
-        range.setEnd(textNode, activeMention.start);
+        range.setStart(textNode, mentionStart);
+        range.setEnd(textNode, mentionStart);
 
         const mentionRect = range.getBoundingClientRect();
 
-        // restore selection
-        range.setStart(textNode, activeMention.end);
-        range.setEnd(textNode, activeMention.end);
+        // restore cursor position
+        range.setStart(textNode, currentCursor);
+        range.setEnd(textNode, currentCursor);
 
         return {
             getBoundingClientRect: () => mentionRect,
         };
     };
 
+    updatePopupReference = (): void => {
+        const { editorState } = this.state;
+
+        const activeMention = getActiveMentionForEditorState(editorState);
+
+        this.setState({ popupReference: activeMention ? this.getVirtualElement(activeMention) : null });
+    };
+
+    handleFirstUpdate = (state: Partial<PopperState>): void => {
+        const { x, y } = state.rects?.reference || {};
+
+        // sometimes inner popper position is wrong due to parent popper position change, like zoom in/out
+        // the inner popper position should always be positive (bottom-right)
+        // so if position is negative after first positioning,
+        // we re-calculate virtualElement to re-position inner popper
+        if ((x && x < 0) || (y && y < 0)) {
```

Comment:
It doesn't seem like this should be needed, since popper has overflow prevention built-in. Can we separate this change from this PR so we can discuss it further?

Reply:
> It doesn't seem like this should be needed, since popper has overflow prevention built-in. Can we separate this change from this PR so we can discuss it further?

sure